app/src/main/java/org/mozilla/focus/widget/BrowserNavigationOverlay.kt

```diff
@@ -105,6 +105,13 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         autocompleteFilter.load(context.applicationContext)
         setOnFilterListener { searchText, view -> autocompleteFilter.onFilter(searchText, view) }
 
+        // #548: Get the current URL when we leave resign focus
+        setOnFocusChangeListener { _, hasFocus ->
+            if (!hasFocus) {
+                setText(navigationStateProvider?.getCurrentUrl())
```

Comment:
Since the url bar is the first thing focused when this screen is shown, we don't set the text when the overlay is initially shown: we should do that in `onStart()`